File: N2.5/N2.5.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strings"
)

func WordConverter(fourthblank string) []string {
	convertedWord := []string{}
	blankWord := fourthblank
	blankWordSlice := strings.Split(blankWord, "")
	sort.Strings(blankWordSlice)
	convertedWord = append(convertedWord, strings.Join(blankWordSlice, ""))
	for {
		i := len(blankWordSlice) - 2
		for i >= 0 && blankWordSlice[i] >= blankWordSlice[i+1] {
			i--
		}
		if i < 0 {
			break
		}
		j := len(blankWordSlice) - 1
		for blankWordSlice[j] <= blankWordSlice[i] {
			j--
		}
		blankWordSlice[i], blankWordSlice[j] = blankWordSlice[j], blankWordSlice[i]
		left := i + 1
		right := len(blankWordSlice) - 1
		for left < right {
			blankWordSlice[left], blankWordSlice[right] = blankWordSlice[right], blankWordSlice[left]
			left++
			right--
		}
		convertedWord = append(convertedWord, strings.Join(blankWordSlice, ""))
	}
	return convertedWord
}

func StringSplitter(enterString string) []string {
	splittedString := strings.Split(enterString, " ")
	return splittedString
}

func Translate(wordToTranslate string, dictionary map[string]string) string {
	translatedWord := ""
	for _, char := range wordToTranslate {
		translatedWord += dictionary[string(char)]
	}
	return translatedWord
}

func UnicChecker(translatedAllCombinations []string, rnComb string) string {
	scoreOfSames := 0
	for _, word := range translatedAllCombinations {
		if rnComb == word {
			scoreOfSames++
		}
	}
	if scoreOfSames > 1 {
		return ""
	} else {
		return rnComb
	}
}

func main() {
	dictionary := map[string]string{
		"a": ".-", "b": "-...", "w": ".--", "g": "--.", "d": "-..",
		"e": ".", "v": "...-", "z": "--..", "i": "..", "j": ".---",
		"k": "-.-", "l": ".-..", "m": "--", "n": "-.", "o": "---",
		"p": ".--.", "r": ".-.", "s": "...", "t": "-", "u": "..-",
		"f": "..-.", "h": "....", "c": "-.-.", "q": "--.-", "y": "-.--", "x": "-..-",
	}

	scanner := bufio.NewScanner(os.Stdin)
	fmt.Print("Enter String - ")
	scanner.Scan()
	enterString := scanner.Text()

	splittedString := StringSplitter(enterString)

	boolCheckOfWordLength := true
	for _, word := range splittedString {
		if len(word) > 9 {
			boolCheckOfWordLength = false
			break
		}
	}

	boolCheckOfStringLength := true
	if len(enterString) > 100 || len(enterString) < 1 {
		boolCheckOfStringLength = false
	}

	if !boolCheckOfStringLength || !boolCheckOfWordLength {
		fmt.Println("Please, enter correct string")
	} else {
		allCombinations := []string{}
		for _, word := range splittedString {
			for _, combination := range WordConverter(word) {
				allCombinations = append(allCombinations, combination)
			}
		}

		translatedAllCombinations := []string{}
		for _, wordToTranslate := range allCombinations {
			translatedAllCombinations = append(translatedAllCombinations, Translate(wordToTranslate, dictionary))
		}

		unicCombinations := []string{}
		for _, rnComb := range translatedAllCombinations {
			if result := UnicChecker(translatedAllCombinations, rnComb); result != "" {
				unicCombinations = append(unicCombinations, result)
			}
		}

		fmt.Println()
		fmt.Println("All combinations and their Morze translation:")
		for i := 0; i < len(allCombinations); i++ {
			fmt.Printf("%s = (%s)\n", allCombinations[i], translatedAllCombinations[i])
		}

		fmt.Println("Unic words:")
		for _, output := range unicCombinations {
			fmt.Printf("(%s) ", output)
		}
		fmt.Println()
		fmt.Println("____________________________________________________")
		fmt.Printf("Value of unic words - %d\n", len(unicCombinations))
	}
}

File: N2.5/N.2.5.py
from itertools import permutations

def word_converter(word):
    sorted_word = ''.join(sorted(word))
    return set(''.join(p) for p in permutations(sorted_word))

def string_splitter(enter_string):
    return enter_string.split()

def translate(word_to_translate, dictionary):
    return ''.join(dictionary[char] for char in word_to_translate)

def unic_checker(translated_all_combinations, rn_comb):
    return rn_comb if translated_all_combinations.count(rn_comb) == 1 else ""

def main():
    dictionary = {
        "a": ".-", "b": "-...", "w": ".--", "g": "--.", "d": "-..",
        "e": ".", "v": "...-", "z": "--..", "i": "..", "j": ".---",
        "k": "-.-", "l": ".-..", "m": "--", "n": "-.", "o": "---",
        "p": ".--.", "r": ".-.", "s": "...", "t": "-", "u": "..-",
        "f": "..-.", "h": "....", "c": "-.-.", "q": "--.-", "y": "-.--", "x": "-..-"
    }

    while True:
        print("Enter String - ")
        enter_string = input()

        if len(enter_string) > 100 or len(enter_string) < 1 or any(len(word) > 9 for word in enter_string.split()):
            print("Please, enter correct string")
            continue

        all_combinations = set()
        for word in string_splitter(enter_string):
            all_combinations.update(word_converter(word))

        translated_all_combinations = [translate(word, dictionary) for word in all_combinations]

        unic_combinations = set(unic_checker(translated_all_combinations, comb) for comb in translated_all_combinations)
        unic_combinations.discard("")

        print("\nAll combinations and their Morze translation: ")
        for word, morze in zip(all_combinations, translated_all_combinations):
            print(f"{word} = ({morze})")

        print("Unic words: ")
        for output in unic_combinations:
            print(f"({output}) ", end="")

        print(f"\n\nValue of unic words - {len(unic_combinations)}")
        break

if __name__ == "__main__":
    main()